### PR TITLE
Add guards for network storage

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

<https://github.com/home-assistant/supervisor/pull/4301> Introduces blocking the API of network mounts, this PR aims to better handle that in the UI.

This PR also includes unrelated changes for mount options, which is planed for a new select default mount dialog (which requires <https://github.com/home-assistant/frontend/pull/16567>)

If the host feature is missing and its running HAOS a warning about needing to update is added
image

If host feature is missing on supervised, anything could be wrong, so they get a simpler message, with no action item.
image

CC @matthiasdebaat 

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/storage/dialog-mount-view.ts [hunk 1]
```diff
@@ -114,13 +129,25 @@ class ViewMountDialog extends LitElement {
...
+    this._data =
+      dialogParams.mount !== undefined
+        ? {
+            ...dialogParams.mount,
+            default_backup_mount:
+              dialogParams.defaultBackupMount !== null
+                ? dialogParams.mount.name === dialogParams.defaultBackupMount
+                : undefined,
+          }
+        : undefined;
```
Comment (bramkragten) [inline src/panels/config/storage/dialog-mount-view.ts:148]: Can we do this logic in a memoize function to get the data in render? So we don't have to add and delete this everywhere and use `_defaultBackupMount` as source of truth?

File: src/panels/config/storage/dialog-mount-view.ts [hunk 2]
```diff
@@ -225,17 +254,41 @@ class ViewMountDialog extends LitElement {
...
+    ) {
+      this._data!.default_backup_mount = true;
+    }
   }
 
   private async _connectMount() {
     this._error = undefined;
     this._waiting = true;
+    const updateDefaultBackupMount = this._data!.default_backup_mount;
+    delete this._data!.default_backup_mount;
```
Comment (bramkragten) [inline src/panels/config/storage/dialog-mount-view.ts:270]: if the updating or creating errors, we will now lose this

File: src/panels/config/storage/dialog-mount-view.ts (hunk 1 above)
Comment (ludeeus) [reply src/panels/config/storage/dialog-mount-view.ts:148]: Does it matter? We only run this one time when you open the dialog anyway?

File: src/panels/config/storage/dialog-mount-view.ts (hunk 2 above)
Comment (ludeeus) [reply src/panels/config/storage/dialog-mount-view.ts:270]: Meh, yeah...
Will adjust it 👍

**Comment by matthiasdebaat** [comment]
We can drop the star, when there is only one backup network storage.

**Comment by ludeeus** [comment]
Looks like it will be dropped always

**Review by bramkragten — APPROVED** [review]

**Pull request merged**